src/request/selectors/RequestDetailDataSelectors.ts

```diff
@@ -1,19 +1,73 @@
 import { IRequestState } from '../stores/IRequestState';
 
 import { createSelector } from 'reselect';
+import * as _ from 'lodash';
 
+export const getFilterState = (state: IRequestState) => state.detail.data.filters;
 export const getOperations = (state: IRequestState) => state.detail.data.operations;
-export const getSelectedIndex = (state: IRequestState) => state.detail.data.selectedIndex;
+export const getSelectedOperationIdState = (state: IRequestState) => state.detail.data.selectedOperationId;
 
 export const getTotalOperationCount = createSelector(
     getOperations,
     operations => {
         return operations.length;
     });
 
+export const getFilteredOperations = createSelector(
+    getFilterState,
+    getOperations,
+    (filterState, operations) => {
+        const hiddenDatabases = _.keys(_.omitBy(filterState, isShown => isShown));
+        
+        let filteredOperations = [];
+        
+        operations.forEach((operation, index) => {
+            if (!_.includes(hiddenDatabases, operation.database)) {
+                filteredOperations.push({
+                    ordinal: index + 1,
+                    operation: operation
+                })
+            }
+        });
+        
+        return filteredOperations;
+    });
```

Comment:
Yes, since we don't support arbitrary sorting in the UI that's taken care of by the reducer (to keep things simple and not have to introduce an intermediate selector to do the sorting), which allows us to just iterate over the state and include/exclude based on the filters.  If we changed the UI, for example, to allow an ascending or descending sort, then we would probably shift responsibility for sorting to a selector.